tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-on-device-training.yml

```diff
@@ -15,30 +15,30 @@ jobs:
     parameters:
       RunDockerBuildArgs: |
         -o ubuntu20.04 -d gpu -e \
-        -t onnxruntime_ondevicetraining_tests_image \
+        -t onnxruntime_training_apis_tests_image \
         -x " \
           --config RelWithDebInfo \
           --enable_training \
-          --enable_training_on_device \
+          --enable_training_apis \
```

Comment:
Considering that enable_training will enable enable_training_apis as well, shall we remove this flag from all the ymls that are currently using this?

Reply:
Also, considering enable_training will also enable enable_trianing_apis, shall we get rid of two CI pipelines and build only once?

Reply:
I intentionally kept enable_training_apis in this PR... In the next PR where I enable creating a dedicated build for training apis , I am deleting --enable_training flag from this CI yml.

Reply:
Do we need a dedicated build for training apis? Shall we just reuse the training CI pipeline which now also builds the training apis?

Reply:
we need to test we are able to build with just enable_training_apis and need to run the relevant tests to validate the builds. Office will be building with just enable_training_apis flag.

Reply:
enable_training_apis will be the way to build for learning on the edge scenarios... Office will be first to use this.